components/shopify/sources/new-event/new-event.js

```diff
@@ -25,27 +29,42 @@ module.exports = {
           summary: event.message,
           ts: Date.now(),
         });
-        if (results[results.length - 1])
-          this.db.set(eventType, results[results.length - 1].id);
       }
+      if (results[results.length - 1])
+        this.db.set(eventType, results[results.length - 1].id);
     },
+    async getEvents(eventType, sinceId) {
+      const results = await this.shopify.getEvents(
+        sinceId,
+        JSON.parse(eventType).filter,
+        JSON.parse(eventType).verb
+      );
+      return { results, eventType };
```

Comment:
Any reason you need to return `eventType` since the caller should have access to it (i.e. since it's passed as a param)?

Reply:
I see what you're doing below now, that makes sense and that's fine.